### PR TITLE
New: babel-polyfill. All dependencies have been updated.

### PR DESCRIPTION
**Pull request by Richard1984**
Solved core-js error: #465 
All dependencies have been updated to solve some security problems.

### CODE REVIEW

**Comment by Richard1984** [comment]
# Packages

## New

* **@babel**/polyfill
* consolidate
* eslint-plugin-import
* eslint-plugin-vue
* eslint-plugin-node
* mini-css-extract-plugin
* optimize-css-assets-webpack-plugin
* uglifyjs-webpack-plugin

## Updated

* **@babel**/cli
* **@babel**/core
* **@babel**/preset-env
* **@babel**/runtime
* babel-eslint
* babel-loader
* babel-plugin-minify-dead-code-elimination
* babel-plugin-transform-inline-environment-variables
* cross-env
* css-loader
* eslint
* eslint-config-standard
* eslint-plugin-html
* eslint-plugin-promise
* eslint-plugin-standard
* file-loader
* gh-pages
* lab
* puppeteer
* raw-loader
* shx
* style-loader
* template-html-loader
* vue
* vue-loader
* vue-router
* vue-template-compiler
* webpack
* webpack-cli

## Removed

* **@babel**/plugin-proposal-object-rest-spread
* babel-preset-es2015
* babel-preset-stage-2
* code
* jsdom
* less
* less-loader
* lodash
* lodash-es
* lodash-webpack-plugin
* node-sass
* sass-loader
* stylus-loader
* vue-hot-reload-api
* vue-html-loader

# Files

## New

|  File |  Description  |
| --- | --- |
|  config/webpack.production.config.js  |  Webpack configuration (Production) |
|  config/webpack.development.config.js  |  Webpack configuration (Development) |
|  config/webpack.test.config.js  |  Webpack configuration (Test) |

# Problems

* **58 security problems:** _solved_
* **Issue #465:** _solved_

# More

* **Files optimization on build:**

	_1. mini-css-extract-plugin
	2. optimize-css-assets-webpack-plugin
	3. uglifyjs-webpack-plugin_

**Comment by Richard1984** [comment]
What about this? https://github.com/Richard1984/vue-gmaps
Using vue-cli 3. @xkjyeah

**Comment by xkjyeah** [comment]
Do you think you can remove `dist/*` to keep the PR clean?

**Comment by Richard1984** [comment]
Do you mean the content of dist or the folder itself?

**Comment by xkjyeah** [comment]
the changes you've introduced to `dist`. I'll update that when publishing a new version

**Comment by Richard1984** [comment]
Sorry, I don't understand. :( Are you referring  to this PR or the other one?

> What about this? https://github.com/Richard1984/vue-gmaps
> Using vue-cli 3. @xkjyeah 

Because the content of *dist* folder in this PR is the same of the original repo.

**Comment by adekoyejoakinhanmi** [comment]
@xkjyeah Is this ever gonna be merged? More so how can I avoid the error this solves before this is finally merged?

**Comment by diegoazh** [comment]
Please resolve this conflicts:
README.md
dist/vue-google-maps.js
package-lock.json
package.json
If you resolve this the I think the PR can be merged.

**Comment by ricristian** [comment]
It seems that this PR won't be merged too soon

**Review by diegoazh — COMMENTED** [review]
This changes looks right for me, I do not have any comment about that. Any one have comments on this PR?

**Comment by diegoazh** [comment]
@Richard1984 can you solve the last conflicts and remove the changes in `dist/` folder, that @xkjyeah request you, I will like to merge this PR on the branch master as soon as posible. Thank you.

**Review by diegoazh — CHANGES_REQUESTED** [review]
Resolve conflicts and remove changes on `dist/` folder in order to be merged.

**Review by diegoazh — APPROVED** [review]

**Comment by diegoazh** [comment]
Thank you, @Richard1984.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vue2-google-maps",
-  "version": "0.10.4",
+  "version": "0.10.2",
```
Comment (diegoazh) [inline package.json]: Why you change the version from 0.10.4 to 0.10.2?

**Review by diegoazh — CHANGES_REQUESTED** [review]
Please upgrade the version in `package.json` accordingly I think this is not a braking change but is a important change because that I think it should be 0.11.0, but if you have doubts leave it as it is 0.10.4 and @xkjyeah will upgrade in a new released.

File: package.json (hunk 1 above)
Comment (diegoazh) [reply package.json]: Please se de package-lock.json too, after fix this I will merge this PR @Richard1984.

**Comment by diegoazh** [comment]
Excellent work!!! @Richard1984

**Review by diegoazh — APPROVED** [review]

**Comment by JoseGoncalves** [comment]
Hi @diegoazh and @Richard1984, first of all thanks for keeping this library alive and updated.

Nevertheless, there is a change in this PR that I do not agree.
You have moved @babel/runtime to devDependencies. As the name of the package implies, is to be used in runtime, i.e., is to be set in dependencies and not in devDependencies. See [@babel/plugin-transform-runtime](https://babeljs.io/docs/en/babel-plugin-transform-runtime) documentation for extra info.

We can question if @babel/runtime package is at all needed, and it will not if you use vue2-google-maps in a webpack environment, but, if it's used directly in an old browser (e.g. IE), vue2-google-maps may stop work with this change.

So my point is, @babel/runtime should be removed if support for old browsers whithout a build environment like webpack is not desired or should be included in the vue2-google-maps dependencies otherwise.

**Comment by diegoazh** [comment]
@JoseGoncalves you are right. I missed this detail. Can you create a new PR with this changes?

**Comment by JoseGoncalves** [comment]
@diegoazh Sure! But in each way should we go, i.e., should we remove @babel/runtime package or should we move it to dependencies?

**Comment by diegoazh** [comment]
@JoseGoncalves is there any way to keep these two possibilities or they are mutually exclusive?

**Comment by JoseGoncalves** [comment]
They are mutually exclusive.
Has you have @babel/runtime set right now (in devDependencies) is exactly the same has not having it at all (but should be removed for consistency).
If it was my call, I would opt for removing that package. I don't think that, in the current days, someone is building Vue apps without a build system like Webpack.
In that case you don't need @babel/runtime, even to run your app in old browsers, because the build system handles the poyfills and transpilation for you.

**Comment by diegoazh** [comment]
@JoseGoncalves is correct, is very difficult to think someone developing a vuejs app without webpack, please remove it.

**Pull request merged**